includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1338,8 +1341,34 @@ private function is_valid_layout( $tag_spec, $node ) {
 			return true;
 		}
 
-		// Elements without a width or height don't need to be validated.
-		if ( ! $node->hasAttribute( 'width' ) && ! $node->hasAttribute( 'height' ) ) {
+		$layout_attr  = $node->getAttribute( Attribute::LAYOUT );
+		$input_width  = new CssLength( $node->getAttribute( Attribute::WIDTH ) );
+		$input_height = new CssLength( $node->getAttribute( Attribute::HEIGHT ) );
+
+		if ( ! $input_width->isDefined() && ! $input_height->isDefined() ) {
+			/*
+			 * Special case. If no layout related attributes were provided, this implies
+			 * the CONTAINER layout. However, telling the user that the implied layout
+			 * is unsupported for this tag is confusing if all they need is to provide
+			 * width and height in, for example, the common case of creating
+			 * an AMP-IMG without specifying dimensions. In this case, we emit a
+			 * less correct, but simpler error message that could be more useful to
+			 * the average user.
+			 *
+			 * See https://github.com/ampproject/amp-wp/issues/4465
+			 */
+			if (
+				'' === $layout_attr
+				&& ! $this->supports_layout( $tag_spec, Layout::CONTAINER )
+				&& $this->supports_layout( $tag_spec, Layout::RESPONSIVE )
```

Comment:
What about when the layout doesn't support responsive either? Is it missing this case here:

https://github.com/ampproject/amphtml/blob/19f1b72d7d57205539f842cddbad97730cfdeee3/validator/engine/validator.js#L4386-L4389

Reply:
Some examples of bad markup not currently caught:

```html
<amp-gist data-gistid="b9bb35bc68df68259af94430f012425f"></amp-gist>
<amp-animation><script type="application/json">{}</script></amp-animation>
```

Curiously, I found that `amp-audio` does not require `layout` and absent `width`/`height` it is no validation error, even though it does _not_ support the `CONTAINER` layout:

```html
<amp-audio src="https://example.com/foo.mp3"></amp-audio>
```

Another component, `amp-pixel`, supports `nodisplay` and `fixed` layouts, but neither are need to be supplied, as this is valid:

```html
<amp-pixel src="https://example.com/tracker/foo"></amp-pixel>
```

Reply:
Fixes for these have been added to this PR.